### ISSUE TITLE
Inference example

### ISSUE THREAD

**Issue by robmarkcole** [IsaacCorley/pytorch-enhance#6]
I assumed I could do the following:
```
# Training with pl

# save for use in production environment
torch.save(model.state_dict(), 'ESPCN.pth')

# Load and evaluate model:
model = ESPCN(scale_factor, channels)
model.load_state_dict(torch.load("ESPCN.pth"))
model.eval()

test_images, test_labels = next(iter(test_dataloader)
lr_img = test_images[0].squeeze()
hr_img = test_labels[0].squeeze()

sr_img = model.enhance(test_images[0])
```
but this results in `RuntimeError: Expected 4-dimensional input for 4-dimensional weight [64, 1, 5, 5], but got 3-dimensional input of size [1, 200, 200] instead` - I'm using `enhance` incorrectly?

**Comment by isaaccorley** [comment]
@robmarkcole What is the shape of test_images[0]? PyTorch models require a batch dimension so I would think that's the issue. Try doing sr_img = model.enhance(test_images[0].unsqueeze(0)).

**Comment by robmarkcole** [comment]
```
test_images[0].shape

torch.Size([1, 200, 200])
```
What is required is `torch.Size([1, 1, 200, 200]` so I try:
```
sr_img = model.enhance(torch.unsqueeze(test_images[0], 0))
``` 
getting 
```
---> 80         x = x.transpose((1, 2, 0))
     81         return x

TypeError: transpose() received an invalid combination of arguments - got (tuple), but expected one of:
 * (name dim0, name dim1)
 * (int dim0, int dim1)
 * 
 ```

Might be a bug [here](https://github.com/isaaccorley/pytorch-enhance/blob/c341a19a3480433eb41ac83be25ac557f1f97219/torch_enhance/models/base.py#L80)? Pretty sure the transpose is not required

In [this example](https://github.com/pytorch/examples/blob/master/super_resolution/super_resolve.py) they are using `input = img_to_tensor(y).view(1, -1, y.size[1], y.size[0])` which achieves the same result..?

**Comment by isaaccorley** [comment]
Yeah it should actually be x = x.permute(1, 2, 0). I think the valid inputs of transpose may have changed over time. I'll make a fix for this and for checking that your input dims are correct as well.

**Comment by isaaccorley** [comment]
Fixed in #7 and #8. Removed the permute/transpose op since this breaks for batch of images. I was mostly using this to just quickly upres a single image and format it to be easily plotted but I think this is a better compromise.
